sdk/maps/Azure.Maps.Render/src/Models/PushpinStyle.cs

```diff
@@ -0,0 +1,170 @@
+        /// </summary>
+        public Uri? CustomPinImage { get; set; }
+
+        /// <summary>
+        /// The anchor location of label for built-in pushpins is at the top center
+        /// of custom pushpins. To override the anchor location of the pin image,
+        /// user can designate how to shift or move the anchor location by pixels
+        /// </summary>
+        public Point? LabelAnchorShiftInPixels { get; set; }
+
+        /// <summary> Label color information. Opacity value other than 1 be ignored. </summary>
+        public Color? LabelColor { get; set; }
+
+        /// <summary>
+        /// Label scale, should greater than 0. A value of 1 is the standard scale.
+        /// Values larger than 1 will make the label larger.
+        /// </summary>
+        public double? LabelScale { get; set; }
+
+        /// <summary>
+        /// A number of degrees of clockwise rotation.
+        /// Use a negative number to rotate counter-clockwise.
+        /// Value can be -360 to 360.
+        /// </summary>
+        public int? RotationInDegrees { get; set; }
+
+        /// <summary> Convert PushpinStyle to endpoint-specific string format. </summary>
+        public override string ToString() {
+            StringBuilder sb = new StringBuilder(256);
+
+            if (CustomPinImage == null)
+            {
+                sb.Append("default");
+            }
+            else
+            {
+                sb.Append("custom");
+            }
+
+            if (PinColor != null)
+            {
+                sb.AppendFormat(CultureInfo.InvariantCulture, "|co{0:X2}{1:X2}{2:X2}",
+                    PinColor?.R, PinColor?.G, PinColor?.A);
+
+                if (PinColor?.A != 255)
+                {
+                    double alpha = Convert.ToInt32(PinColor?.A, CultureInfo.InvariantCulture) / 255.0;
+                    sb.AppendFormat(CultureInfo.InvariantCulture, "|al{0:0.###}", alpha);
+                }
+            }
+
+            if (PinScale != null)
+            {
+                sb.AppendFormat(CultureInfo.InvariantCulture, "|sc{0:0.##}", PinScale);
+            }
+
+            if (PinAnchorShiftInPixels != null)
+            {
+                sb.Append($"|an{PinAnchorShiftInPixels?.X} {PinAnchorShiftInPixels?.Y}");
+            }
+
+            if (RotationInDegrees != null)
+            {
+                sb.Append($"|ro{RotationInDegrees}");
+            }
+
+            if (LabelColor != null)
+            {
+                sb.AppendFormat(CultureInfo.InvariantCulture, "|lc{0:X2}{1:X2}{2:X2}",
+                    LabelColor?.R, LabelColor?.G, LabelColor?.B);
+            }
+
+            if (LabelScale != null)
+            {
+                sb.AppendFormat(CultureInfo.InvariantCulture, "|ls{0:0.##}", LabelScale);
+            }
+
+            if (LabelAnchorShiftInPixels != null)
+            {
+                sb.Append($"|la{LabelAnchorShiftInPixels?.X} {LabelAnchorShiftInPixels?.Y}");
```

Comment:
I beleive using a bling string here is going to cause unnecessary allocations.   I'd suggest either doing these as individual `Append` calls or using `AppendFormat`.   Likewise for the below.